### PR TITLE
fix: brazilob jarm fapi conformance test last7 issues

### PR DESCRIPTION
**Pull request by HemantKMehta**
properly formatted code fix for Brazil OB JARM fapi1-advanced-final issue (new numbers on monorepo)# 310 to 315

### CODE REVIEW

File: jans-auth-server/client/src/test/java/io/jans/as/client/BaseTest.java [hunk 1]
```diff
@@ -1162,7 +1163,42 @@ public RegisterResponse registerClient(
         assertNotNull(registerResponse.getClientSecret());
         assertNotNull(registerResponse.getRegistrationAccessToken());
         assertNotNull(registerResponse.getClientIdIssuedAt());
-        assertNotNull(registerResponse.getClientSecretExpiresAt());
+        //assertNotNull(registerResponse.getClientSecretExpiresAt());
```
Comment (yuriyz) [inline jans-auth-server/client/src/test/java/io/jans/as/client/BaseTest.java]: Normally expiration should be present, no ? At least for clients which we dynamically registered by tests.

File: jans-auth-server/client/src/test/java/io/jans/as/client/BaseTest.java [hunk 2]
```diff
@@ -1162,7 +1163,42 @@ public RegisterResponse registerClient(
...
+        registerClient.setRequest(registerRequest);
+        RegisterResponse registerResponse = registerClient.exec();
+
+        showClient(registerClient);
+        assertEquals(registerResponse.getStatus(), 201, "Unexpected response code: " + registerResponse.getEntity());
+        assertNotNull(registerResponse.getClientId());
+        assertNotNull(registerResponse.getClientSecret());
+        assertNotNull(registerResponse.getRegistrationAccessToken());
+        assertNotNull(registerResponse.getClientIdIssuedAt());
+        //assertNotNull(registerResponse.getClientSecretExpiresAt());
```
Comment (yuriyz) [inline jans-auth-server/client/src/test/java/io/jans/as/client/BaseTest.java]: same thing

File: jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeJwtResponseTypeCodeSignedEncryptedHttpTest.java [hunk 3]
```diff
@@ -39,9 +42,598 @@
...
+    public void userDeniedAuthentiation(
+        final String redirectUri, final String redirectUris, final String clientJwksUri,
+        final String encryptionKeyId, final String signingKeyId, final String dnName, final String keyStoreFile,
+        final String keyStoreSecret, final String sectorIdentifierUri) throws Exception {
+	    showTitle("userDeniedAuthentiation");
+	
+	    
+	
+	    // 1. Dynamic Client Registration
+	    //RegisterResponse registerResponse = registerClient(redirectUris, responseTypes, sectorIdentifierUri, clientJwksUri,
```
Comment (yuriyz) [inline jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeJwtResponseTypeCodeSignedEncryptedHttpTest.java]: Many new lines (55-56) as well as commented lines in this class. Lets remove unused code.

File: jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeJwtResponseTypeCodeSignedEncryptedHttpTest.java [hunk 4]
```diff
@@ -39,9 +42,598 @@
...
+	    String serverKeyId = jwkResponse.getKeyId(Algorithm.RSA_OAEP);
+	    assertNotNull(serverKeyId);
+	
+	    JSONObject jwks = JwtUtil.getJSONWebKeys(jwksUri);
+	    AuthCryptoProvider cryptoProvider2 = new AuthCryptoProvider(keyStoreFile, keyStoreSecret, dnName);
+	    privateKey = cryptoProvider2.getPrivateKey(encryptionKeyId);
+	    
+        
+	  
+	    //JwtAuthorizationRequest jweAuthorizationRequest = new JwtAuthorizationRequest(authorizationRequest,
```
Comment (yuriyz) [inline jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeJwtResponseTypeCodeSignedEncryptedHttpTest.java]: Unused code and redundant new lines.

File: jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeJwtResponseTypeCodeSignedEncryptedHttpTest.java [hunk 5]
```diff
@@ -39,9 +42,598 @@
...
+	 
+	    Jwe response = Jwe.parse(authorizationResponse.getResponse(), privateKey, null);
+	    assertNotNull(response.getClaims().getClaimAsString(AuthorizeResponseParam.ISS));
+	    assertNotNull(response.getClaims().getClaimAsString(AuthorizeResponseParam.AUD));
+	    assertNotNull(response.getClaims().getClaimAsInteger(AuthorizeResponseParam.EXP));
+	    assertNotNull(response.getClaims().getClaimAsString(AuthorizeResponseParam.STATE));
+	    assertNull(response.getClaims().getClaimAsString(AuthorizeResponseParam.CODE));
+	    //assertNotNull(response.getClaims().getClaimAsString("error"));
+	    //assertNotNull(response.getClaims().getClaimAsString("error_description"));
+	    //assertEquals(response.getClaims().getClaimAsString("error"), "invalid_request_object");
```
Comment (yuriyz) [inline jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/jarm/AuthorizationResponseModeJwtResponseTypeCodeSignedEncryptedHttpTest.java]: If you wish to assert error, simply add it to https://github.com/JanssenProject/jans/blob/546b1c23becbf9d00cdb4aaaeb086a44a9fb518b/jans-auth-server/client/src/test/java/io/jans/as/client/client/Asserter.java#L38

No need to keep commented lines.

File: jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java [hunk 6]
```diff
@@ -395,7 +399,25 @@ private Response requestAuthorization(
                     }
                     requestParameterService.getCustomParameters(jwtRequest, customParameters);
                 } catch (WebApplicationException e) {
-                    throw e;
+					responseMode = extractResponseMode(request);
```
Comment (yuriyz) [inline jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java]: Would you please replace tabs with spaces? Then it does not look messed on github.

File: jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java [hunk 7]
```diff
@@ -395,7 +399,25 @@ private Response requestAuthorization(
...
+					if (responseMode == ResponseMode.JWT) {
+						JwtClaims jwtClaims = Jwt.parseSilently(request).getClaims();
+						fillRedirectUriResponseforJARM(redirectUriResponse, clientId, client);
```
Comment (yuriyz) [inline jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java]: 1. Inside `fillRedirectUriResponseforJARM` method we are passing clientId and client objects. Client has that id already thus no need to pass `clientId` (`client.getClientId()`).
2. pass inside method Jwt object, and populate whatever is needed inside: 

```
private void fillRedirectUriResponseforJARM(redirectUriResponse, jwt, client) {
    JwtClaims jwtClaims = jwt.getClaims();
    String tempRedirectUri = jwtClaims.getClaimAsString("redirect_uri");
    if (tempRedirectUri != null) {
        redirectUriResponse.getRedirectUri().setBaseRedirectUri(URLDecoder.decode(tempRedirectUri, "UTF-8"));
    }
    redirectUriResponse.getRedirectUri().setResponseMode(ResponseMode.JWT);
    state = jwtClaims.getClaimAsString("state");
    if (state != null) {
        redirectUriResponse.setState(state);
    }
    ....
}
```

File: jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java [hunk 8]
```diff
@@ -807,7 +778,78 @@ private Response requestAuthorization(
...
     }
-
+    
+	private ResponseMode extractResponseMode(String request) {
+		JwtClaims jwtClaims = Jwt.parseSilently(request).getClaims();
+		return ResponseMode.getByValue(jwtClaims.getClaimAsString("response_mode"));
+	}
+    private void fillRedirectUriResponseforJARM(RedirectUriResponse redirectUriResponse, String clientId, Client client) {
+    	try {
+			redirectUriResponse.getRedirectUri().setIssuer(appConfiguration.getIssuer());
```
Comment (yuriyz) [inline jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java]: tabs->spaces

File: jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceValidator.java [hunk 9]
```diff
@@ -179,7 +179,13 @@ public void validateRequestObject(JwtAuthorizationRequest jwtRequest, RedirectUr
                 log.error("The Nested JWT signature algorithm is not valid.");
                 throw redirectUriResponse.createWebException(AuthorizeErrorResponseType.INVALID_REQUEST_OBJECT);
             }
-        }
+        }      
+		String redirectUri = jwtRequest.getRedirectUri();
```
Comment (yuriyz) [inline jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceValidator.java]: tabs -> spaces

**Review by yuriyz — CHANGES_REQUESTED** [review]
Logic itself is good but code needs to be improved a bit. Please check inlined comments.

File: jans-auth-server/client/src/test/java/io/jans/as/client/BaseTest.java (hunk 1 above)
Comment (HemantKMehta) [reply jans-auth-server/client/src/test/java/io/jans/as/client/BaseTest.java]: done

**Pull request closed without merging**